src/core/highlight.js

```diff
@@ -45,32 +45,44 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
-        }
-        clearTimeout(timeoutId);
-        worker.removeEventListener("message", listener);
-        done();
+      const elementsToHighlight =
+        element.querySelectorAll("code").length > 0
+          ? Array.from(element.querySelectorAll("code"))
+          : [element];
+      elementsToHighlight.map((each, id) => {
+        const msg = {
+          action: "highlight",
+          code: each.innerText,
+          id: `highlight:${id}`,
+          languages: getLanguageHint(each.classList),
+        };
+        worker.addEventListener("message", function listener(ev) {
+          const {
+            data: { id, language, value },
+          } = ev;
+          if (id !== msg.id) {
+            return; // not for us!
+          }
+          const lang = language !== undefined ? language : msg.languages[0];
+          switch (each.localName) {
+            case "pre":
+              each.innerHTML = `<code class="${lang}">${value}</code>`;
+              each.classList.remove(lang);
+              each.firstChild.classList.add("hljs");
```

Comment:
I think this needs to stay on the `pre`, not the firstChild. Note also, `firstChild` will match text nodes, so try to be specific (i.e., `firstElementNode` when you want to target an element).

Reply:
I guess the firstChild here will only be `<code>` since, I am wrapping the complete `<pre>` into `<code>`.